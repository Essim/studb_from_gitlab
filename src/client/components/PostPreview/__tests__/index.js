import React from 'react';
import {PostPreview} from '../';
import { shallowToJson } from 'enzyme-to-json';

const mockDocument={
    _id: '5a17f595977b356a3d6d392a',
    cuid: 'cuidDoc1',
    cuid_course: 'anglais-1',
    cuid_division: 'division-bin',
    cuid_author: 'cuidusr2',
    name_author: 'christopher',
    isRep: true,
    parentRep: null,
    logicalDelete: false,
    name: 'synthese_alphabet',
    description: 'The amazing alphabet',
    date: '2000-12-31T23:00:00.000Z',
    path: 'http://localhost:8888/files/version_doc1',
    zip_path: 'http://localhost:8888/files/version_doc1/version_doc1.zip',
    __v: 0,
    tagsNumber: [
        1,
        2,
        3
    ],
    tags: [
        {
            name: 'BIN',
            division: true,
            course: false,
            logicalDelete: false,
            numid: 1,
            cuid: 'cuidTag1',
            _id: '5a17f595977b356a3d6d392d',
        },
        {
            name: 'Anglais',
            division: false,
            course: true,
            logicalDelete: false,
            numid: 2,
            cuid: 'cuidTag2',
            _id: '5a17f595977b356a3d6d392c',
        },
        {
            name: 'Funny',
            division: false,
            course: false,
            logicalDelete: false,
            numid: 3,
            cuid: 'cuidTag3',
            _id: '5a17f595977b356a3d6d392b',
        }
    ],
    comments: [
        {
            cuid_author: 'cuidusr2',
            name_author: 'christopher',
            text: 'Superbe test',
            date: '2017-11-06T14:01:15.000Z',
            mainComment: true,
            logicalDelete: false,
            cuid: 'cuidCom1',
            _id: '5a17f595977b356a3d6d3930',
            opinions: [
                {
                    cuid_author: 'cuidusr2',
                    name_author: 'christopher',
                    date: '2001-12-31T23:00:00.000Z',
                    state: true,
                    report: false,
                    logicalDelete: false,
                    cuid: 'opi3',
                    _id: '5a17f595977b356a3d6d3931'
                }
            ],
            comments: [
                {
                    cuid_author: 'cuidusr1',
                    name_author: 'essim',
                    text: 'One step test',
                    date: '2004-12-31T23:00:00.000Z',
                    mainComment: false,
                    logicalDelete: false,
                    cuid: 'cuidCom2',
                    _id: '5a17f595977b356a3d6d3932',
                    opinions: [],
                    comments: []
                }
            ]
        },
        {
            cuid_author: 'cuidusr1',
            name_author: 'essim',
            text: 'Je suis une banane',
            date: '2017-11-06T14:01:15.000Z',
            mainComment: true,
            logicalDelete: false,
            cuid: 'cuidCom2',
            _id: '5a17f595977b356a3d6d392e',
            opinions: [
                {
                    cuid_author: 'cuidusr2',
                    name_author: 'christopher',
                    date: '2001-12-31T23:00:00.000Z',
                    state: true,
                    report: false,
                    logicalDelete: false,
                    cuid: 'opi3',
                    _id: '5a17f595977b356a3d6d392f'
                }
            ],
            comments: []
        }
    ],
    opinions: [
        {
            cuid_author: 'cuidusr1',
            name_author: 'essim',
            date: '2001-12-31T23:00:00.000Z',
            state: true,
            report: false,
            logicalDelete: false,
            cuid: 'opi1',
            _id: '5a17f595977b356a3d6d3934'
        },
        {
            cuid_author: 'cuidusr2',
            name_author: 'christopher',
            date: '2001-12-31T23:00:00.000Z',
            state: true,
            report: false,
            logicalDelete: false,
            cuid: 'opi2',
            _id: '5a17f595977b356a3d6d3933'
        }
    ],
    files: {
        dir1: {
            _contents: [
                {
                    name: 'file3.docx',
                    isDir: false,
                    path: 'files/version_doc1/dir1/file3.docx',
                    url: 'http://localhost:8888/files/version_doc1/dir1/file3.docx',
                    Ext: '.docx'
                }
            ]
        },
        _contents: [
            {
                name: 'file1.pdf',
                isDir: false,
                path: 'files/version_doc1/file1.pdf',
                url: 'http://localhost:8888/files/version_doc1/file1.pdf',
                Ext: '.pdf'
            },
            {
                name: 'file2.doc',
                isDir: false,
                path: 'files/version_doc1/file2.doc',
                url: 'http://localhost:8888/files/version_doc1/file2.doc',
                Ext: '.doc'
            }
        ]
    }
},mockDivisions= [
    {
        _id: '5a17f595977b356a3d6d3918',
        acronym: 'BIN',
        cuid: 'cuiddivision1',
        logicalDelete: false,
        __v: 0,
        courses: [
            {
                name: 'Anglais',
                grade: 1,
                logicalDelete: false,
                cuid: 'anglais-1',
                _id: '5a17f595977b356a3d6d3920'
            },
            {
                name: 'Web',
                grade: 1,
                logicalDelete: false,
                cuid: 'web-1',
                _id: '5a17f595977b356a3d6d391f'
            },
            {
                name: 'Assembleur',
                grade: 1,
                logicalDelete: false,
                cuid: 'assembly-1',
                _id: '5a17f595977b356a3d6d391e'
            },
            {
                name: 'Anglais',
                grade: 2,
                logicalDelete: false,
                cuid: 'anglais-2',
                _id: '5a17f595977b356a3d6d391d'
            },
            {
                name: 'Web',
                grade: 2,
                logicalDelete: false,
                cuid: 'web-2',
                _id: '5a17f595977b356a3d6d391c'
            },
            {
                name: 'Language C',
                grade: 2,
                logicalDelete: false,
                cuid: 'c-lang-1',
                _id: '5a17f595977b356a3d6d391b'
            },
            {
                name: 'Anglais',
                grade: 3,
                logicalDelete: false,
                cuid: 'anglais-3',
                _id: '5a17f595977b356a3d6d391a'
            },
            {
                name: 'Web',
                grade: 3,
                logicalDelete: false,
                cuid: 'web-3',
                _id: '5a17f595977b356a3d6d3919'
            }
        ],
        grades: [
            null,
            {
                name: '1BIN',
                grade: 1,
                courses: [
                    {
                        name: 'Anglais',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'anglais-1',
                        _id: '5a17f595977b356a3d6d3920'
                    },
                    {
                        name: 'Web',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'web-1',
                        _id: '5a17f595977b356a3d6d391f'
                    },
                    {
                        name: 'Assembleur',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'assembly-1',
                        _id: '5a17f595977b356a3d6d391e'
                    }
                ]
            },
            {
                name: '2BIN',
                grade: 2,
                courses: [
                    {
                        name: 'Anglais',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'anglais-2',
                        _id: '5a17f595977b356a3d6d391d'
                    },
                    {
                        name: 'Web',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'web-2',
                        _id: '5a17f595977b356a3d6d391c'
                    },
                    {
                        name: 'Language C',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'c-lang-1',
                        _id: '5a17f595977b356a3d6d391b'
                    }
                ]
            },
            {
                name: '3BIN',
                grade: 3,
                courses: [
                    {
                        name: 'Anglais',
                        grade: 3,
                        logicalDelete: false,
                        cuid: 'anglais-3',
                        _id: '5a17f595977b356a3d6d391a'
                    },
                    {
                        name: 'Web',
                        grade: 3,
                        logicalDelete: false,
                        cuid: 'web-3',
                        _id: '5a17f595977b356a3d6d3919'
                    }
                ]
            }
        ]
    },
    {
        _id: '5a17f595977b356a3d6d3921',
        acronym: 'BIM',
        cuid: 'division-bim',
        logicalDelete: false,
        __v: 0,
        courses: [
            {
                name: 'Anglais',
                grade: 1,
                logicalDelete: false,
                cuid: 'anglais-1',
                _id: '5a17f595977b356a3d6d3929'
            },
            {
                name: 'Web',
                grade: 1,
                logicalDelete: false,
                cuid: 'web-1',
                _id: '5a17f595977b356a3d6d3928'
            },
            {
                name: 'Assembleur',
                grade: 1,
                logicalDelete: false,
                cuid: 'assembly-1',
                _id: '5a17f595977b356a3d6d3927'
            },
            {
                name: 'Anglais',
                grade: 2,
                logicalDelete: false,
                cuid: 'anglais-2',
                _id: '5a17f595977b356a3d6d3926'
            },
            {
                name: 'Web',
                grade: 2,
                logicalDelete: false,
                cuid: 'web-2',
                _id: '5a17f595977b356a3d6d3925'
            },
            {
                name: 'Language C',
                grade: 2,
                logicalDelete: false,
                cuid: 'c-lang-1',
                _id: '5a17f595977b356a3d6d3924'
            },
            {
                name: 'Anglais',
                grade: 3,
                logicalDelete: false,
                cuid: 'anglais-3',
                _id: '5a17f595977b356a3d6d3923'
            },
            {
                name: 'Web',
                grade: 3,
                logicalDelete: false,
                cuid: 'web-3',
                _id: '5a17f595977b356a3d6d3922'
            }
        ],
        grades: [
            null,
            {
                name: '1BIM',
                grade: 1,
                courses: [
                    {
                        name: 'Anglais',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'anglais-1',
                        _id: '5a17f595977b356a3d6d3929'
                    },
                    {
                        name: 'Web',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'web-1',
                        _id: '5a17f595977b356a3d6d3928'
                    },
                    {
                        name: 'Assembleur',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'assembly-1',
                        _id: '5a17f595977b356a3d6d3927'
                    }
                ]
            },
            {
                name: '2BIM',
                grade: 2,
                courses: [
                    {
                        name: 'Anglais',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'anglais-2',
                        _id: '5a17f595977b356a3d6d3926'
                    },
                    {
                        name: 'Web',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'web-2',
                        _id: '5a17f595977b356a3d6d3925'
                    },
                    {
                        name: 'Language C',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'c-lang-1',
                        _id: '5a17f595977b356a3d6d3924'
                    }
                ]
            },
            {
                name: '3BIM',
                grade: 3,
                courses: [
                    {
                        name: 'Anglais',
                        grade: 3,
                        logicalDelete: false,
                        cuid: 'anglais-3',
                        _id: '5a17f595977b356a3d6d3923'
                    },
                    {
                        name: 'Web',
                        grade: 3,
                        logicalDelete: false,
                        cuid: 'web-3',
                        _id: '5a17f595977b356a3d6d3922'
                    }
                ]
            }
        ]
    }
];

describe('PostPreview', () => {
    it('should render correctly', () => {
        const output = shallow(<PostPreview document={mockDocument} divisions={mockDivisions}/>);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
