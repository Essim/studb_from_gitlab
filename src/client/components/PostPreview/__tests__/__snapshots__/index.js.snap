// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPreview should render correctly 1`] = `
<div
  className=""
>
  <div
    className="col-xs-undefined col-xs-offset-undefined"
  >
    <div
      className="box box-widget"
    >
      <div
        className="box-header with-border"
      >
        <div
          className="user-block"
        >
          <Gravatar
            className="img-circle"
            default="identicon"
            email="christopher"
            protocol="//"
            rating="g"
            size={50}
          />
          <span
            className="username"
          >
            <Link
              replace={false}
              to="/user/stats/christopher"
            >
              christopher
            </Link>
          </span>
          <span
            className="description"
          >
            <i>
              Cours inconnu
            </i>
             -
            Le 2001-1-1 Ã  00:00:00
          </span>
          <span
            className="comment"
          >
             
            <a
              key="linktocuidTag1"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              #
              BIN
            </a>
             
            <a
              key="linktocuidTag2"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              #
              Anglais
            </a>
             
            <a
              key="linktocuidTag3"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              #
              Funny
            </a>
          </span>
        </div>
        <div
          className="box-tools"
        >
          <button
            className="btn btn-box-tool"
            data-toggle="tooltip"
            title="Consulter"
            type="button"
          >
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "/document/cuidDoc1",
                  "state": Object {
                    "document": Object {
                      "__v": 0,
                      "_id": "5a17f595977b356a3d6d392a",
                      "comments": Array [
                        Object {
                          "_id": "5a17f595977b356a3d6d3930",
                          "comments": Array [
                            Object {
                              "_id": "5a17f595977b356a3d6d3932",
                              "comments": Array [],
                              "cuid": "cuidCom2",
                              "cuid_author": "cuidusr1",
                              "date": "2004-12-31T23:00:00.000Z",
                              "logicalDelete": false,
                              "mainComment": false,
                              "name_author": "essim",
                              "opinions": Array [],
                              "text": "One step test",
                            },
                          ],
                          "cuid": "cuidCom1",
                          "cuid_author": "cuidusr2",
                          "date": "2017-11-06T14:01:15.000Z",
                          "logicalDelete": false,
                          "mainComment": true,
                          "name_author": "christopher",
                          "opinions": Array [
                            Object {
                              "_id": "5a17f595977b356a3d6d3931",
                              "cuid": "opi3",
                              "cuid_author": "cuidusr2",
                              "date": "2001-12-31T23:00:00.000Z",
                              "logicalDelete": false,
                              "name_author": "christopher",
                              "report": false,
                              "state": true,
                            },
                          ],
                          "text": "Superbe test",
                        },
                        Object {
                          "_id": "5a17f595977b356a3d6d392e",
                          "comments": Array [],
                          "cuid": "cuidCom2",
                          "cuid_author": "cuidusr1",
                          "date": "2017-11-06T14:01:15.000Z",
                          "logicalDelete": false,
                          "mainComment": true,
                          "name_author": "essim",
                          "opinions": Array [
                            Object {
                              "_id": "5a17f595977b356a3d6d392f",
                              "cuid": "opi3",
                              "cuid_author": "cuidusr2",
                              "date": "2001-12-31T23:00:00.000Z",
                              "logicalDelete": false,
                              "name_author": "christopher",
                              "report": false,
                              "state": true,
                            },
                          ],
                          "text": "Je suis une banane",
                        },
                      ],
                      "cuid": "cuidDoc1",
                      "cuid_author": "cuidusr2",
                      "cuid_course": "anglais-1",
                      "cuid_division": "division-bin",
                      "date": "2000-12-31T23:00:00.000Z",
                      "description": "The amazing alphabet",
                      "files": Object {
                        "_contents": Array [
                          Object {
                            "Ext": ".pdf",
                            "isDir": false,
                            "name": "file1.pdf",
                            "path": "files/version_doc1/file1.pdf",
                            "url": "http://localhost:8888/files/version_doc1/file1.pdf",
                          },
                          Object {
                            "Ext": ".doc",
                            "isDir": false,
                            "name": "file2.doc",
                            "path": "files/version_doc1/file2.doc",
                            "url": "http://localhost:8888/files/version_doc1/file2.doc",
                          },
                        ],
                        "dir1": Object {
                          "_contents": Array [
                            Object {
                              "Ext": ".docx",
                              "isDir": false,
                              "name": "file3.docx",
                              "path": "files/version_doc1/dir1/file3.docx",
                              "url": "http://localhost:8888/files/version_doc1/dir1/file3.docx",
                            },
                          ],
                        },
                      },
                      "isRep": true,
                      "logicalDelete": false,
                      "name": "synthese_alphabet",
                      "name_author": "christopher",
                      "opinions": Array [
                        Object {
                          "_id": "5a17f595977b356a3d6d3934",
                          "cuid": "opi1",
                          "cuid_author": "cuidusr1",
                          "date": "2001-12-31T23:00:00.000Z",
                          "logicalDelete": false,
                          "name_author": "essim",
                          "report": false,
                          "state": true,
                        },
                        Object {
                          "_id": "5a17f595977b356a3d6d3933",
                          "cuid": "opi2",
                          "cuid_author": "cuidusr2",
                          "date": "2001-12-31T23:00:00.000Z",
                          "logicalDelete": false,
                          "name_author": "christopher",
                          "report": false,
                          "state": true,
                        },
                      ],
                      "parentRep": null,
                      "path": "http://localhost:8888/files/version_doc1",
                      "tags": Array [
                        Object {
                          "_id": "5a17f595977b356a3d6d392d",
                          "course": false,
                          "cuid": "cuidTag1",
                          "division": true,
                          "logicalDelete": false,
                          "name": "BIN",
                          "numid": 1,
                        },
                        Object {
                          "_id": "5a17f595977b356a3d6d392c",
                          "course": true,
                          "cuid": "cuidTag2",
                          "division": false,
                          "logicalDelete": false,
                          "name": "Anglais",
                          "numid": 2,
                        },
                        Object {
                          "_id": "5a17f595977b356a3d6d392b",
                          "course": false,
                          "cuid": "cuidTag3",
                          "division": false,
                          "logicalDelete": false,
                          "name": "Funny",
                          "numid": 3,
                        },
                      ],
                      "tagsNumber": Array [
                        1,
                        2,
                        3,
                      ],
                      "zip_path": "http://localhost:8888/files/version_doc1/version_doc1.zip",
                    },
                  },
                }
              }
            >
              <i
                className="fa fa-expand"
              />
            </Link>
          </button>
        </div>
      </div>
      <div
        className="box-body with-border"
      >
        <Link
          replace={false}
          to={
            Object {
              "pathname": "/document/cuidDoc1",
              "state": Object {
                "document": Object {
                  "__v": 0,
                  "_id": "5a17f595977b356a3d6d392a",
                  "comments": Array [
                    Object {
                      "_id": "5a17f595977b356a3d6d3930",
                      "comments": Array [
                        Object {
                          "_id": "5a17f595977b356a3d6d3932",
                          "comments": Array [],
                          "cuid": "cuidCom2",
                          "cuid_author": "cuidusr1",
                          "date": "2004-12-31T23:00:00.000Z",
                          "logicalDelete": false,
                          "mainComment": false,
                          "name_author": "essim",
                          "opinions": Array [],
                          "text": "One step test",
                        },
                      ],
                      "cuid": "cuidCom1",
                      "cuid_author": "cuidusr2",
                      "date": "2017-11-06T14:01:15.000Z",
                      "logicalDelete": false,
                      "mainComment": true,
                      "name_author": "christopher",
                      "opinions": Array [
                        Object {
                          "_id": "5a17f595977b356a3d6d3931",
                          "cuid": "opi3",
                          "cuid_author": "cuidusr2",
                          "date": "2001-12-31T23:00:00.000Z",
                          "logicalDelete": false,
                          "name_author": "christopher",
                          "report": false,
                          "state": true,
                        },
                      ],
                      "text": "Superbe test",
                    },
                    Object {
                      "_id": "5a17f595977b356a3d6d392e",
                      "comments": Array [],
                      "cuid": "cuidCom2",
                      "cuid_author": "cuidusr1",
                      "date": "2017-11-06T14:01:15.000Z",
                      "logicalDelete": false,
                      "mainComment": true,
                      "name_author": "essim",
                      "opinions": Array [
                        Object {
                          "_id": "5a17f595977b356a3d6d392f",
                          "cuid": "opi3",
                          "cuid_author": "cuidusr2",
                          "date": "2001-12-31T23:00:00.000Z",
                          "logicalDelete": false,
                          "name_author": "christopher",
                          "report": false,
                          "state": true,
                        },
                      ],
                      "text": "Je suis une banane",
                    },
                  ],
                  "cuid": "cuidDoc1",
                  "cuid_author": "cuidusr2",
                  "cuid_course": "anglais-1",
                  "cuid_division": "division-bin",
                  "date": "2000-12-31T23:00:00.000Z",
                  "description": "The amazing alphabet",
                  "files": Object {
                    "_contents": Array [
                      Object {
                        "Ext": ".pdf",
                        "isDir": false,
                        "name": "file1.pdf",
                        "path": "files/version_doc1/file1.pdf",
                        "url": "http://localhost:8888/files/version_doc1/file1.pdf",
                      },
                      Object {
                        "Ext": ".doc",
                        "isDir": false,
                        "name": "file2.doc",
                        "path": "files/version_doc1/file2.doc",
                        "url": "http://localhost:8888/files/version_doc1/file2.doc",
                      },
                    ],
                    "dir1": Object {
                      "_contents": Array [
                        Object {
                          "Ext": ".docx",
                          "isDir": false,
                          "name": "file3.docx",
                          "path": "files/version_doc1/dir1/file3.docx",
                          "url": "http://localhost:8888/files/version_doc1/dir1/file3.docx",
                        },
                      ],
                    },
                  },
                  "isRep": true,
                  "logicalDelete": false,
                  "name": "synthese_alphabet",
                  "name_author": "christopher",
                  "opinions": Array [
                    Object {
                      "_id": "5a17f595977b356a3d6d3934",
                      "cuid": "opi1",
                      "cuid_author": "cuidusr1",
                      "date": "2001-12-31T23:00:00.000Z",
                      "logicalDelete": false,
                      "name_author": "essim",
                      "report": false,
                      "state": true,
                    },
                    Object {
                      "_id": "5a17f595977b356a3d6d3933",
                      "cuid": "opi2",
                      "cuid_author": "cuidusr2",
                      "date": "2001-12-31T23:00:00.000Z",
                      "logicalDelete": false,
                      "name_author": "christopher",
                      "report": false,
                      "state": true,
                    },
                  ],
                  "parentRep": null,
                  "path": "http://localhost:8888/files/version_doc1",
                  "tags": Array [
                    Object {
                      "_id": "5a17f595977b356a3d6d392d",
                      "course": false,
                      "cuid": "cuidTag1",
                      "division": true,
                      "logicalDelete": false,
                      "name": "BIN",
                      "numid": 1,
                    },
                    Object {
                      "_id": "5a17f595977b356a3d6d392c",
                      "course": true,
                      "cuid": "cuidTag2",
                      "division": false,
                      "logicalDelete": false,
                      "name": "Anglais",
                      "numid": 2,
                    },
                    Object {
                      "_id": "5a17f595977b356a3d6d392b",
                      "course": false,
                      "cuid": "cuidTag3",
                      "division": false,
                      "logicalDelete": false,
                      "name": "Funny",
                      "numid": 3,
                    },
                  ],
                  "tagsNumber": Array [
                    1,
                    2,
                    3,
                  ],
                  "zip_path": "http://localhost:8888/files/version_doc1/version_doc1.zip",
                },
              },
            }
          }
        >
          <h2>
            synthese_alphabet
          </h2>
        </Link>
        <MarkdownArea
          content="The amazing alphabet"
        />
      </div>
      <Connect(FilePreview)
        doc_cuid="cuidDoc1"
        selected_file={
          Object {
            "Ext": ".pdf",
            "isDir": false,
            "name": "file1.pdf",
            "path": "files/version_doc1/file1.pdf",
            "url": "http://localhost:8888/files/version_doc1/file1.pdf",
          }
        }
      />
      <div
        className="box-body with-border"
      >
        <Opinions
          comments={
            Array [
              Object {
                "_id": "5a17f595977b356a3d6d3930",
                "comments": Array [
                  Object {
                    "_id": "5a17f595977b356a3d6d3932",
                    "comments": Array [],
                    "cuid": "cuidCom2",
                    "cuid_author": "cuidusr1",
                    "date": "2004-12-31T23:00:00.000Z",
                    "logicalDelete": false,
                    "mainComment": false,
                    "name_author": "essim",
                    "opinions": Array [],
                    "text": "One step test",
                  },
                ],
                "cuid": "cuidCom1",
                "cuid_author": "cuidusr2",
                "date": "2017-11-06T14:01:15.000Z",
                "logicalDelete": false,
                "mainComment": true,
                "name_author": "christopher",
                "opinions": Array [
                  Object {
                    "_id": "5a17f595977b356a3d6d3931",
                    "cuid": "opi3",
                    "cuid_author": "cuidusr2",
                    "date": "2001-12-31T23:00:00.000Z",
                    "logicalDelete": false,
                    "name_author": "christopher",
                    "report": false,
                    "state": true,
                  },
                ],
                "text": "Superbe test",
              },
              Object {
                "_id": "5a17f595977b356a3d6d392e",
                "comments": Array [],
                "cuid": "cuidCom2",
                "cuid_author": "cuidusr1",
                "date": "2017-11-06T14:01:15.000Z",
                "logicalDelete": false,
                "mainComment": true,
                "name_author": "essim",
                "opinions": Array [
                  Object {
                    "_id": "5a17f595977b356a3d6d392f",
                    "cuid": "opi3",
                    "cuid_author": "cuidusr2",
                    "date": "2001-12-31T23:00:00.000Z",
                    "logicalDelete": false,
                    "name_author": "christopher",
                    "report": false,
                    "state": true,
                  },
                ],
                "text": "Je suis une banane",
              },
            ]
          }
          document={
            Object {
              "__v": 0,
              "_id": "5a17f595977b356a3d6d392a",
              "comments": Array [
                Object {
                  "_id": "5a17f595977b356a3d6d3930",
                  "comments": Array [
                    Object {
                      "_id": "5a17f595977b356a3d6d3932",
                      "comments": Array [],
                      "cuid": "cuidCom2",
                      "cuid_author": "cuidusr1",
                      "date": "2004-12-31T23:00:00.000Z",
                      "logicalDelete": false,
                      "mainComment": false,
                      "name_author": "essim",
                      "opinions": Array [],
                      "text": "One step test",
                    },
                  ],
                  "cuid": "cuidCom1",
                  "cuid_author": "cuidusr2",
                  "date": "2017-11-06T14:01:15.000Z",
                  "logicalDelete": false,
                  "mainComment": true,
                  "name_author": "christopher",
                  "opinions": Array [
                    Object {
                      "_id": "5a17f595977b356a3d6d3931",
                      "cuid": "opi3",
                      "cuid_author": "cuidusr2",
                      "date": "2001-12-31T23:00:00.000Z",
                      "logicalDelete": false,
                      "name_author": "christopher",
                      "report": false,
                      "state": true,
                    },
                  ],
                  "text": "Superbe test",
                },
                Object {
                  "_id": "5a17f595977b356a3d6d392e",
                  "comments": Array [],
                  "cuid": "cuidCom2",
                  "cuid_author": "cuidusr1",
                  "date": "2017-11-06T14:01:15.000Z",
                  "logicalDelete": false,
                  "mainComment": true,
                  "name_author": "essim",
                  "opinions": Array [
                    Object {
                      "_id": "5a17f595977b356a3d6d392f",
                      "cuid": "opi3",
                      "cuid_author": "cuidusr2",
                      "date": "2001-12-31T23:00:00.000Z",
                      "logicalDelete": false,
                      "name_author": "christopher",
                      "report": false,
                      "state": true,
                    },
                  ],
                  "text": "Je suis une banane",
                },
              ],
              "cuid": "cuidDoc1",
              "cuid_author": "cuidusr2",
              "cuid_course": "anglais-1",
              "cuid_division": "division-bin",
              "date": "2000-12-31T23:00:00.000Z",
              "description": "The amazing alphabet",
              "files": Object {
                "_contents": Array [
                  Object {
                    "Ext": ".pdf",
                    "isDir": false,
                    "name": "file1.pdf",
                    "path": "files/version_doc1/file1.pdf",
                    "url": "http://localhost:8888/files/version_doc1/file1.pdf",
                  },
                  Object {
                    "Ext": ".doc",
                    "isDir": false,
                    "name": "file2.doc",
                    "path": "files/version_doc1/file2.doc",
                    "url": "http://localhost:8888/files/version_doc1/file2.doc",
                  },
                ],
                "dir1": Object {
                  "_contents": Array [
                    Object {
                      "Ext": ".docx",
                      "isDir": false,
                      "name": "file3.docx",
                      "path": "files/version_doc1/dir1/file3.docx",
                      "url": "http://localhost:8888/files/version_doc1/dir1/file3.docx",
                    },
                  ],
                },
              },
              "isRep": true,
              "logicalDelete": false,
              "name": "synthese_alphabet",
              "name_author": "christopher",
              "opinions": Array [
                Object {
                  "_id": "5a17f595977b356a3d6d3934",
                  "cuid": "opi1",
                  "cuid_author": "cuidusr1",
                  "date": "2001-12-31T23:00:00.000Z",
                  "logicalDelete": false,
                  "name_author": "essim",
                  "report": false,
                  "state": true,
                },
                Object {
                  "_id": "5a17f595977b356a3d6d3933",
                  "cuid": "opi2",
                  "cuid_author": "cuidusr2",
                  "date": "2001-12-31T23:00:00.000Z",
                  "logicalDelete": false,
                  "name_author": "christopher",
                  "report": false,
                  "state": true,
                },
              ],
              "parentRep": null,
              "path": "http://localhost:8888/files/version_doc1",
              "tags": Array [
                Object {
                  "_id": "5a17f595977b356a3d6d392d",
                  "course": false,
                  "cuid": "cuidTag1",
                  "division": true,
                  "logicalDelete": false,
                  "name": "BIN",
                  "numid": 1,
                },
                Object {
                  "_id": "5a17f595977b356a3d6d392c",
                  "course": true,
                  "cuid": "cuidTag2",
                  "division": false,
                  "logicalDelete": false,
                  "name": "Anglais",
                  "numid": 2,
                },
                Object {
                  "_id": "5a17f595977b356a3d6d392b",
                  "course": false,
                  "cuid": "cuidTag3",
                  "division": false,
                  "logicalDelete": false,
                  "name": "Funny",
                  "numid": 3,
                },
              ],
              "tagsNumber": Array [
                1,
                2,
                3,
              ],
              "zip_path": "http://localhost:8888/files/version_doc1/version_doc1.zip",
            }
          }
          document_cuid="cuidDoc1"
          file_path="http://localhost:8888/files/version_doc1.zip"
          isZip={true}
          key="OpinionscuidDoc1"
          opinions={
            Array [
              Object {
                "_id": "5a17f595977b356a3d6d3934",
                "cuid": "opi1",
                "cuid_author": "cuidusr1",
                "date": "2001-12-31T23:00:00.000Z",
                "logicalDelete": false,
                "name_author": "essim",
                "report": false,
                "state": true,
              },
              Object {
                "_id": "5a17f595977b356a3d6d3933",
                "cuid": "opi2",
                "cuid_author": "cuidusr2",
                "date": "2001-12-31T23:00:00.000Z",
                "logicalDelete": false,
                "name_author": "christopher",
                "report": false,
                "state": true,
              },
            ]
          }
        />
      </div>
      <Connect(BrowserFiles)
        doc_cuid="cuidDoc1"
        files={
          Object {
            "_contents": Array [
              Object {
                "Ext": ".pdf",
                "isDir": false,
                "name": "file1.pdf",
                "path": "files/version_doc1/file1.pdf",
                "url": "http://localhost:8888/files/version_doc1/file1.pdf",
              },
              Object {
                "Ext": ".doc",
                "isDir": false,
                "name": "file2.doc",
                "path": "files/version_doc1/file2.doc",
                "url": "http://localhost:8888/files/version_doc1/file2.doc",
              },
            ],
            "dir1": Object {
              "_contents": Array [
                Object {
                  "Ext": ".docx",
                  "isDir": false,
                  "name": "file3.docx",
                  "path": "files/version_doc1/dir1/file3.docx",
                  "url": "http://localhost:8888/files/version_doc1/dir1/file3.docx",
                },
              ],
            },
          }
        }
      />
      <div
        className="box-footer"
      >
        <a
          className="text-center"
          href="#"
          onClick={[Function]}
        >
          Afficher les commentaires
        </a>
      </div>
    </div>
  </div>
</div>
`;
